# Ejercicios en clase de pytorch

In [14]:
pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121

Looking in indexes: https://download.pytorch.org/whl/cu121
  Using cached https://download.pytorch.org/whl/cu121/torch-2.5.1%2Bcu121-cp312-cp312-win_amd64.whl (2449.3 MB)
  Using cached https://download.pytorch.org/whl/cu121/torchvision-0.20.1%2Bcu121-cp312-cp312-win_amd64.whl (6.1 MB)
     ---------------------------------------- 0.0/4.1 MB ? eta -:--:--
     -- ------------------------------------- 0.3/4.1 MB ? eta -:--:--
     ----- ---------------------------------- 0.5/4.1 MB 1.9 MB/s eta 0:00:02
     ---------- ----------------------------- 1.0/4.1 MB 1.8 MB/s eta 0:00:02
     ------------ --------------------------- 1.3/4.1 MB 1.8 MB/s eta 0:00:02
     --------------- ------------------------ 1.6/4.1 MB 1.8 MB/s eta 0:00:02
     -------------------- ------------------- 2.1/4.1 MB 1.8 MB/s eta 0:00:02
     ---------------------- ----------------- 2.4/4.1 MB 1.8 MB/s eta 0:00:01
     --------------------------- ------------ 2.9/4.1 MB 1.8 MB/s eta 0:00:01
     --------------------

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
cirq-core 1.6.1 requires pandas~=2.1, but you have pandas 3.0.1 which is incompatible.
sentence-transformers 5.1.2 requires transformers<5.0.0,>=4.41.0, but you have transformers 5.3.0 which is incompatible.


In [1]:
import torch
import torch.nn as nn

autograd es calcular la distribucion de pesos por neuronas

optimizadores es el gradient descent para actualizar los pesos


In [2]:
import torch 
import numpy as np
x= torch.tensor([1,2,3])

array = np.array([1,2,3])
print(array) 
print(x)

x1 = torch.tensor(array)
print(x1)

[1 2 3]
tensor([1, 2, 3])
tensor([1, 2, 3])


In [3]:
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print(device)

cuda


In [4]:
import torch

print(torch.cuda.is_available())
print(torch.cuda.device_count())
print(torch.version.cuda)

True
1
12.1


Creo que en este kernel de Python no tengo cuda, instalar

In [5]:
import torch
print(torch.version.cuda)

12.1


In [6]:
!nvcc --version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Wed_Aug_20_13:58:20_Pacific_Daylight_Time_2025
Cuda compilation tools, release 13.0, V13.0.88
Build cuda_13.0.r13.0/compiler.36424714_0


In [7]:
!nvidia-smi

Thu Mar 12 18:26:58 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 591.74                 Driver Version: 591.74         CUDA Version: 13.1     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 5060 ...  WDDM  |   00000000:02:00.0 Off |                  N/A |
| N/A   36C    P4             11W /   50W |       0MiB /   8151MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

El problema era que instale pytorch sin cuda, ya instale el pytorch con cuda 12.1 y ahora si agarro mi device